File: backend/src/main/java/com/backend/backend/friendrequest/FriendRequest.java
package com.backend.backend.friendrequest;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "friend_requests", uniqueConstraints = {@UniqueConstraint(columnNames = {"from_user_id", "to_user_id"})})
public class FriendRequest {

    public enum Status {
        PENDING,
        ACCEPTED,
        REJECTED
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "from_user_id", nullable = false)
    private Long fromUserID;

    @Column(name = "to_user_id", nullable = false)
    private Long toUserID;

    @Column(name = "status", nullable = false)
    @Enumerated(EnumType.STRING)
    private Status status;
}

File: fightme_webapp/lib/Models/friend_request.dart
enum Status { pending, accepted, rejected }

class FriendRequest {
  int id = 0;
  int fromUserID = 0;
  int toUserID = 0;
  Status status = Status.pending;

  FriendRequest.empty() {
    id = 0;
    fromUserID = 0;
    toUserID = 0;
    status = Status.pending;
  }

  bool isEmpty() {
    return id == 0 &&
        fromUserID == 0 &&
        toUserID == 0 &&
        status == Status.pending;
  }

  FriendRequest.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    fromUserID = json['fromUserID'];
    toUserID = json['toUserID'];
    status = (json['status'] == 'STATUS.PENDING')
        ? Status.pending
        : (json['status'] == 'STATUS.ACCEPTED')
            ? Status.accepted
            : Status.rejected;
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'fromUserID': fromUserID,
        'toUserID': toUserID,
        'status': status.toString().toUpperCase()
      };
}
